### PR TITLE
Skybox and Void

### PR DESCRIPTION
**Pull request by KyroVibe**
### Description
Updated the skybox and revamped the "void". Currently set to a light theme to try out. Skybox has some sky customization but I'm not using it at the moment while we pin down an aesthetic.

** Example images taken in Ultra quality **

Screenshot 2023-06-29 at 10 39 05 AM

Screenshot 2023-06-29 at 10 50 17 AM

### Objectives
- [x] Added skybox sphere mesh to draw sky
- [x] Blend ground into the skybox
- [x] Allowed grid to receive shadows
- [x] Add horizon transition
- [x] Rewrote grid shader to merge with shadow shader
- [x] Added periodic thick lines to grid
- [x] Added grid line feathering
- [x] Switched environment to a light theme to play with
- [x] Reduced number of quality settings available
- [x] Customized quality settings' shadow cascades
- [x] Added temp fix for shadow cascades issue
- [x] Added solid color environment lighting
- [x] Ambient Lighting 

## Testing Done
- Spawned in multiple bots
- Spawned in multiple fields
- Tried all quality settings and switched between them

[JIRA Issue](https://jira.autodesk.com/browse/AARD-1546)

### CODE REVIEW

File: api/Api/Mirabuf/Appearance.cs [hunk 1]
```diff
@@ -36,7 +36,7 @@ public partial class Appearance {
                         _unityMaterial.SetFloat(OPAQUE_METALLIC, (float)Metallic);
                     }
                     
-                    _unityMaterial.renderQueue = 3000;
+                    // _unityMaterial.renderQueue = 3000;
```
Comment (BrandonPacewic) [inline api/Api/Mirabuf/Appearance.cs:39]: Remove this line?

File: engine/Assets/Scripts/Graphics/GridRenderer.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+        // private void Update() {
+        //     RenderParams rp = new RenderParams(Mat);
+        //     rp.worldBounds = new Bounds(Vector3.zero, Vector3.one * 100);
+        //     rp.matProps = new MaterialPropertyBlock();
+        //     rp.matProps.SetBuffer("_Points", _buff);
+        //     UGraphics.RenderPrimitives(rp, MeshTopology.Lines, 2, 1);
+        // }
+    }
+}
```
Comment (BrandonPacewic) [inline engine/Assets/Scripts/Graphics/GridRenderer.cs:36]: Does this file need to exist? It doesn't appear to be used by anything.

**Review by BrandonPacewic — CHANGES_REQUESTED** [review]
Going to need to add a light mode, my eyes hurt.

File: engine/Assets/Scripts/Camera/FreeCameraMode.cs [hunk 3]
```diff
@@ -125,7 +125,7 @@ public class FreeCameraMode : ICameraMode
     }
     
     public void LateUpdate(CameraController cam) {
-        cam.GroundRenderer.material.SetVector("FOCUS_POINT", cam.transform.position);
+        cam.GroundRenderer.material.SetVector("_GridFocusPoint", cam.transform.position);
```
Comment (PepperLola) [inline engine/Assets/Scripts/Camera/FreeCameraMode.cs:128]: This is also going to have to change in the driver station camera mode. Might be worth merging in dev? I'm not sure what the best way to handle merging PRs is

**Comment by LucaHaverty** [comment]
image
image

Is this circle a rendering bug?

**Review by LucaHaverty — APPROVED** [review]
A couple things about the graphics I noticed (not necessarily problems with changes in this pr):

- In medium and high quality, the shadows are strange:
(not sure if there is much you can do about that)

image

- The strange circle appearing that I mentioned in another comment
- Sometimes the transparent material is rendered behind objects. I don't know if it is used at the moment, but it would be nice to use it for scoring zones and/or robot positioning
image

Overall it looks a lot better, especially the lighting and shadows. I think most of those problems were already around before these changes. Nice work!

**Review by Markytt — APPROVED** [review]

**Comment by KyroVibe** [comment]
> image image
> 
> Is this circle a rendering bug?

Yes, it's a rendering bug. I waiting to see if anyone on [StackOverflow](https://stackoverflow.com/questions/76578541/unity-shadow-cascades-cast-a-shadow) knows what's going on but I've gotten the effect to a manageable point.

**Comment by KyroVibe** [comment]
@PepperLola I'll investigate the transparent shader issue later today.

File: engine/Assets/Scripts/Camera/FreeCameraMode.cs (hunk 3 above)
Comment (KyroVibe) [reply engine/Assets/Scripts/Camera/FreeCameraMode.cs:128]: It'll be handled when resolving conflicts

**Review by BrandonPacewic — APPROVED** [review]
Next I want dark* mode

**Pull request merged**